holidays/countries/senegal.py

```diff
@@ -0,0 +1,147 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.calendars.gregorian import MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT
+from holidays.groups import ChristianHolidays, IslamicHolidays, InternationalHolidays
+from holidays.holiday_base import HolidayBase
+
+
+class Senegal(HolidayBase, ChristianHolidays, IslamicHolidays, InternationalHolidays):
+    """
+    Senegal holidays.
+
+    Referenes:
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Senegal>
+    """
+
+    country = "SN"
+    default_language = "fr"
+    # %s (estimated).
+    estimated_label = tr("%s (estimé)")
+    start_year = 1960
```

Comment:
Maybe 1961 instead? Considering that they gained independence on Apr 4th, 1960 - unless they got records for semi-independence period like Niger's implementation that is.
```suggestion
    start_year = 1961
```